h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -314,12 +313,18 @@ public void doGeneric(String method, HttpServletRequest request, HttpServletResp
 
       resp.writeTo(response.getOutputStream());
 
+    } catch (IllegalArgumentException e) {  // This error type should be passed to clients.
```

Comment:
@wendycwong @michal-raska the reason why IOException requires special handling is mentioned in the comment "Trying to send an error message or stack trace will produce another IOException...".

I don't think we need similar handling for IllegalArgumentException, however I might just be missing something. Please explain the purpose.